### PR TITLE
Added optional "ui_caching" parameter to Broker.start()

### PR DESCRIPTION
**Pull request by colinbr96**
Controls whether the browser should cache the Parlay UI. If `ui_caching` is set to true, the UI shall remain cached for up to 1 hour before becoming stale. If false, the UI shall never be cached.

### CODE REVIEW

File: parlay/server/broker.py [hunk 1]
```diff
@@ -584,6 +585,17 @@ def run(self, mode=Modes.DEVELOPMENT, ssl_only=False, use_ssl=False, open_browse
             root = static.File(PARLAY_PATH + "/ui/dist")
             root.putChild("docs", static.File(PARLAY_PATH + "/docs/_build/html"))
 
+        # overloading twisted.server.Site.getResourceFor to add HTTP headers for setting browser cache
+        class Site(server.Site):
+            def getResourceFor(self, request):
+                if not ui_caching:
+                    cache_strategy = "no-store, must-revalidate"
+                else:
+                    freshness_time_secs = 3600 # content younger than 1 hour is considered fresh
```
Comment (aaronddavies) [inline parlay/server/broker.py]: Define this value in the module space

Comment (aaronddavies) [reply parlay/server/broker.py]: You can also use inline ifs in python:
cache_strategy = x if y else z

Comment (colinbr96) [reply parlay/server/broker.py]: If I use the ternary operator, it goes over the line limit. So I decided to keep it this way to maintain PEP8

Comment (colinbr96) [reply parlay/server/broker.py]: Also, I refactored out freshness_time_secs to be a module-wide variable in the next commit :)

Comment (aaronddavies) [reply parlay/server/broker.py]: Line continuations are fine too
cache_strategy = x \
    if y \
    else z

**Comment by aaronddavies** [comment]
I finally figured out how to approve it

**Pull request merged**